File: ConsumoElectrico/src/Informe.java
public class Informe {
    //Atributos
    String contenido;
    String fecha;
    String hora;
    String ubicacion;
    String autor;
    String destinatario;
    String datosSensor;
    String datosActuador;
    String datosPiso;
    String datosEdificio;
    //Metodos
    public void enviarInforme() {
        System.out.println("Enviando informe...");
    }

    public void archivarInforme() {
        System.out.println("Archivando informe...");
    }

    public void descargarInforme() {
        System.out.println("Descargando informe...");
    }
    //Constructor
    //Asignar valores a los atributos de la instancia actual
   public void generarInformePiso(String contenido, String fecha, String hora, String ubicacion, String autor, String
           destinatario, String datosSensor, String datosActuador, String datosPiso) {
        this.contenido = contenido;
        this.fecha = fecha;
        this.hora = hora;
        this.ubicacion = ubicacion;
        this.autor = autor;
        this.destinatario = destinatario;
        this.datosSensor = datosSensor;
        this.datosActuador = datosActuador;
        this.datosPiso = datosPiso;
    }

    // toma varios parámetros para inicializar los atributos de un objeto
    public void generarInformeEdificio(String contenido, String fecha, String hora, String ubicacion, String autor, String
        destinatario, String datosSensor, String datosActuador, String datosEdificio) {
        this.contenido = contenido;
        this.fecha = fecha;
        this.hora = hora;
        this.ubicacion = ubicacion;
        this.autor = autor;
        this.destinatario = destinatario;
        this.datosSensor = datosSensor;
        this.datosActuador = datosActuador;
        this.datosEdificio = datosEdificio;
    }
}

File: tasks/Templates/static/js/informe.js
function generarInforme() {
    var fechaActual = new Date();
    var hora = fechaActual.getHours();
    var minutos = fechaActual.getMinutes();

    var contenido = document.getElementById("contenido").value;
    var autor = document.getElementById("autor").value;
    var destinatario = document.getElementById("destinatario").value;
    var ubicacion = document.getElementById("ubicacion").value;
    var fecha = document.getElementById("fecha").value;
    var hora = document.getElementById("hora").value;
    var piso = document.getElementById("piso").value;
    var amperios = document.getElementById("amperios").value;
    var horas = document.getElementById("horas").value;

    var datosActuador = "Datos del Actuador: " + (amperios * horas) + " Amperios al día";

    var informeCompleto = "Contenido: " + contenido + "<br>"
        + "Autor: " + autor + "<br>"
        + "Destinatario: " + destinatario + "<br>"
        + "Ubicación: " + ubicacion + "<br>"
        + "Fecha: " + fecha + "<br>"
        + "Hora: " + hora + "<br>"
        + "Piso: " + piso + "<br>"
        + "Datos del Sensor: " + amperios + " Amperios cada " + horas + " horas" + "<br>"
        + datosActuador;

    var informeResult = document.getElementById("informe-result");
    informeResult.innerHTML = informeCompleto;

    document.getElementById("opciones-generar-informe").style.display = "block";
}

function enviarInforme() {
    alert("Enviando informe...");
}

function archivarInforme() {
    alert("Archivando informe...");
}

function descargarInforme() {
    alert("Descargando informe...");
}

function generarInformePiso() {
    alert("Generando informe del piso...");
}

function generarInformeEdificio() {
    alert("Generando informe del edificio...");
}